### PR TITLE
Fix source map domain issue and decoding URI (#328)

### PR DESCRIPTION
**Pull request by jbhoosreddy**
After looking into so many things, it turned out to be easier than I thought. @jasonLaster 
Let me know if this is fine.

### CODE REVIEW

File: public/js/utils/sources-tree.js [hunk 1]
```diff
@@ -59,6 +59,11 @@ function getURL(source) {
       path: "/",
       group: url
     };
+  } else if (urlObj.protocol === "webpack:") {
```
Comment (jasonLaster) [inline public/js/utils/sources-tree.js]: haha - we might want to do something like this or a non http... not sure

File: public/js/utils/sources-tree.js [hunk 2]
```diff
@@ -72,6 +77,7 @@ function addToTree(tree, source) {
   if (!url) {
     return;
   }
+  url.path = decodeURIComponent(url.path);
```
Comment (jasonLaster) [inline public/js/utils/sources-tree.js]: nice, thanks!

**Comment by jasonLaster** [comment]
👍  this solution as is will need to be generalized so that it works in other cases, but for now is probably good

**Comment by jbhoosreddy** [comment]
I didn't know the other cases, so I only aimed at solving the one I knew. But I can definitely make it more comprehensive. I guess we just need to make sure that the parser is robust enough to understand most corner-cases.

File: public/js/utils/sources-tree.js (hunk 1 above)
Comment (jlongster) [inline public/js/utils/sources-tree.js]: It's probably better to whitelist the protocols where know there is a domain/host, at least as a temporary solution. Change it to:

```
} else if (urlObj.protocol === "http" || urlObj.protocol === "https") {
  return {
    path: urlObj.pathname,
    group: urlObj.host
  }
}

return {
  path: urlObj.path,
  group: urlObj.protocol + "//"
}
```

Comment (jlongster) [inline public/js/utils/sources-tree.js]: And what's the difference between `path` and `pathname` again?

**Comment by jlongster** [comment]
A small change to the `getUrl` function will make this work for other things as well, so make that change and this is good to merge.

**Comment by jasonLaster** [comment]
@jlongster what do you mean by `getUrl` function?

File: public/js/utils/sources-tree.js (hunk 1 above)
Comment (jasonLaster) [inline public/js/utils/sources-tree.js]: I believe path is `pathname + search`

**Pull request merged**